model/src/test_manager/mod.rs

```diff
@@ -14,20 +13,45 @@ mod manager;
 mod manager_impl;
 mod status;
 
-/// `SelectionParams` are used to select a group (or single) object from a testsys cluster.
-pub enum SelectionParams {
-    // TODO add field selectors (Think kube-rs `ListParams`)
-    Label(String),
-    Name(CrdName),
-    All,
+#[derive(Default, Debug, Clone)]
+/// `SelectionParams` are used to select a group (or single) object from a testsys cluster. For any
+/// of the filters, None is equivalent to all.
+pub struct SelectionParams {
+    /// Filter based on the type of the CRD
+    pub crd_type: Option<CrdType>,
+    /// Filter based on the crd labels
+    pub labels: Option<String>,
+    /// Filter based on the name of the CRD
+    pub name: Option<String>,
+    /// Filter based on the state of the CRD
+    pub state: Option<CrdState>,
 }
 
-impl Default for SelectionParams {
-    fn default() -> Self {
-        Self::All
-    }
+#[derive(Debug, Clone)]
+/// Filter based on the type of the CRD
+pub enum CrdType {
+    Test,
+    Resource,
 }
 
+#[derive(Debug, Clone, Deserialize)]
+#[serde(rename_all = "kebab-case")]
+/// Filter based on the state of the CRD
+pub enum CrdState {
+    /// All `Test`s and `Resource`s that are not finished.
+    NotFinished,
+    /// All `Test`s and `Resource`s that are currently running.
+    Running,
+    /// All `Test`s and `Resource`s that are finished.
+    Completed,
+    /// All `Test`s that passed.
+    Passed,
+    /// All `Test`s that failed.
+    Failed,
```

Comment:
Hmmm, what does it mean if I say `--resources --state passed`?

Reply:
> It will filter out non-resources and will show all of the tests that passed. (Nothing will be shown).

Is that the best design? What if we made it more generally applicable to Resources and Tests by eliminating the ability to list both passed and failed at the same time...

The issue I'm trying to solve for is a CLI that can take nonsensical input.

```rust
pub enum CrdState {
    /// All `Test`s and `Resource`s that are not finished.
    NotFinished,
    /// All `Test`s and `Resource`s that are currently running.
    Running,
    /// All `Test`s  that passed and `Resource`s that have been successfully created.
    Succeeded,
    /// All `Test`s that failed and `Resource`s that have failed to be created.
    Failed,
}
```

Reply:
I want a user to be able to call `cli status --passed` and only see the passed tests. The user shouldn't care about the state of the resources. Same with `cli status --failed`.We shouldn't require `cli status --test --passed` for the main use case. IMO